### DISCUSSION TITLE
npm cache size increases (over time?) to many GBs

### DISCUSSION THREAD

**Discussion by szpak** [renovatebot/renovate#31366] [Request Help]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab, Renovate 37.273.0

### Please tell us more about your question or problem

We've started using Renovate (on GitLab CI) more for npm projects and after a few months the `others/npm/content-v2` directory has ~33GB (which compression and upload extend the build time by ~10 minutes). I've seen https://github.com/renovatebot/renovate/discussions/13730 and https://github.com/renovatebot/renovate/issues/13732 and the pruning in fact might work as in addition, I see that the files/dirs modification dates are quite recent (a few months). Is it possible to adjust the "expiration date"?

Unfortunately, I don't know how the dir size has been increasing over time (I plan to remove the cache and try to check its size on CI), but I wonder, if there anything else I could do to reduce size of that cache size (the other dirs are reasonable smaller)?

I also use Redis cache, but AFAIR it caches some other data.

**Comment by rarkins** [comment]
This is npm's own cache, i.e. only the `npm` command reads and writes to it. It's usually not safe for any other tool like Renovate to do "surgery" on another tool's cache to prune it - the only safe approach would be to remove it completely at a time when `npm` is not running.

To learn more, you would need to search about npm's cache and how to manage or prune it (the results won't be Renovate-specific). It's possible that the only alternative is to run something like `npm cache clean --force` or a `rm -rf` command directly to remove it.

**Reply by szpak** [reply]
Good point. I will consult with the frontend guys, how they see it. I will check, how it's lack impact the Renovate build time (dependency resolving, etc). Thanks for your tip.
